haddock-api/src/Haddock.hs

```diff
@@ -160,8 +162,9 @@ haddockWithGhc ghc args = handleTopExceptions $ do
     forM_ (warnings args) $ \warning -> do
       hPutStrLn stderr warning
 
+  -- We disable pattern match warnings because than can be very expensive to check
   ghc flags' $ do
-
+    modifySession unsetPatternMatchWarnings
```

Comment:
We already have `withGhc'` at https://github.com/haskell/haddock/blob/master/haddock-api/src/Haddock.hs#L392 would it make sense to place this there? Having all the flag manipulation stuff in one place seems like a good idea.